neurokit2/epochs/eventrelated_utils.py

```diff
@@ -89,8 +89,59 @@ def _eventrelated_rate(epoch, output={}, var="ECG_Rate"):
     zero = find_closest(0, epoch.index.values, return_index=True)  # Find closest to 0
     baseline = epoch[var].iloc[zero]
 
-    signal = epoch[var].values[zero + 1 : :]
-    index = epoch.index.values[zero + 1 : :]
+    # Create smaller epoch within epoch provided when applicable
+    # Sanitize time_start and time_end format
+    if isinstance(time_start, dict):
+        if var not in time_start.keys():
+            time_start = None
+            warn(
+                "`time_start` for `" + var + "` is not provided. "
+                "If it's intentional, set `time_start` for `" + var + "` to None to silent this warning.",
+                category=NeuroKitWarning
+                )
+        else:
+            time_start = time_start[var]
+    if isinstance(time_end, dict):
+        if var not in time_end.keys():
+            time_end = None
+            warn(
+                "`time_end` for `" + var + "` is not provided. "
+                "If it's intentional, set `time_end` for `" + var + "` to None to silent this warning.",
+                category=NeuroKitWarning
+                )
+        else:
+            time_end = time_end[var]
+
+    # Normal behaviour if time_start and time_end are both None
+    if time_start is None and time_end is None:
+        signal = epoch[var].values[zero + 1 : :]
+        index = epoch.index.values[zero + 1 : :]
+    else:
+        # Sanitize time_start and time_end values
+        if time_start is None:
+            time_start = epoch.index.values[zero]  # set time_start to time zero
```

Comment:
@DominiqueMakowski I think it makes sense to set time_start to zero (index where t=0) and not first index of the epoch since in conventional cases (where both arguments are None), signal is from index zero + 1. This is related to my comment regarding cases when it's not convention (line 143)

Reply:
ok